charts/kong/templates/_helpers.tpl

```diff
@@ -439,6 +439,7 @@ The name of the Service which will be used by the controller to update the Ingre
 
   {{- if .Values.ingressController.gatewayDiscovery.enabled -}}
     {{- $_ := set $autoEnv "CONTROLLER_KONG_ADMIN_SVC" (printf "%s/%s-%s" (include "kong.namespace" .) (include "kong.fullname" .) "admin") -}}
+    {{- $_ := set $autoEnv "CONTROLLER_GATEWAY_DISCOVERY_DNS_STRATEGY" "pod" -}}
```

Comment:
Given it's almost impossible to configure mTLS in a reasonable way when using `ip` strategy, what do you think about changing the default on the KIC side as well?

Reply:
My initial response was "I don't know, I don't even know what our use cases were for adding the others; Patryk's doc just said to use `pod` and `pod` does work". Reviewing the actual history there reveals a problem: `pod` apparently [does not work on anything that uses kube-dns (e.g. GKE)](https://github.com/Kong/kubernetes-ingress-controller/issues/4065).

The cert can't list IP SANs because we don't know what they are beforehand, ant KIC#4065 indicates that we can't use `service` either as I read it. It looks like there are are [alternate hostnames](https://github.com/Kong/kubernetes-ingress-controller/issues/4065#issuecomment-1560958169) available under kube-dns, but we didn't implement support for multiple Pod hostname schemes in favor of the simpler IP option, so we'd probably need to revisit that. Would be interested in hearing @pmalek's thoughts when he returns.

Reply:
`pod` actually works fine; we had some incorrect assumptions and blamed kube-dns incorrectly: https://github.com/Kong/kubernetes-ingress-controller/issues/3934

Certificates valid for the entire namespace are still a problem, but arguably better than the status quo in `ingress`, where there's no TLS.

Dunno if there should be a default change because it seems like we maybe shouldn't have a switch at all--there should arguably be a single implementation that can validate arbitrary hostnames.